site/source/docs/debugging/Sanitizers.rst

```diff
@@ -104,3 +104,196 @@ The minimal runtime is supported by Emscripten. To use it, pass the flag
+
+  0x03203e40 is located 0 bytes inside of 400-byte region [0x03203e40,0x03203fd0)
+  freed by thread T0 here:
+      #0 0x5fe8 in operator delete[](void*)+0x5fe8 (a.out.wasm+0x5fe8)
+      #1 0xb76 in __original_main use_after_free.cpp:3:3
+      #2 0xc48 in main+0xc48 (a.out.wasm+0xc48)
+      #3 0x800019b5 in Object.Module._main a.out.js:6581:32
+      #4 0x80001ade in Object.callMain a.out.js:6878:30
+      #5 0x80001b18 in doRun a.out.js:6936:60
+      #6 0x80001b26 in run a.out.js:6950:5
+      #7 0x80001ac9 in runCaller a.out.js:6857:29
+
+  previously allocated by thread T0 here:
+      #0 0x5db4 in operator new[](unsigned long)+0x5db4 (a.out.wasm+0x5db4)
+      #1 0xb41 in __original_main use_after_free.cpp:2:16
+      #2 0xc48 in main+0xc48 (a.out.wasm+0xc48)
+      #3 0x800019b5 in Object.Module._main a.out.js:6581:32
+      #4 0x80001ade in Object.callMain a.out.js:6878:30
+      #5 0x80001b18 in doRun a.out.js:6936:60
+      #6 0x80001b26 in run a.out.js:6950:5
+      #7 0x80001ac9 in runCaller a.out.js:6857:29
+
+  SUMMARY: AddressSanitizer: heap-use-after-free (a.out.wasm+0xc1a)
+  ...
+
+Memory Leaks
+^^^^^^^^^^^^
+
+Consider ``leak.cpp``:
+
+.. code-block:: cpp
+
+  int main() {
+    new int[10];
+  }
+
+Note that since leak checks take place at program exit, you must use
+``-s EXIT_RUNTIME``, or invoke ``__lsan::DoLeakCheck`` manually.
+
+Also, if you only want to check for memory leaks, you may use
+``-fsanitize=leak`` instead of ``-fsanitize=address``.
+
+.. code-block:: console
+
+  $ em++ -g4 -fsanitize=address -s ALLOW_MEMORY_GROWTH -s EXIT_RUNTIME leak.cpp
+  $ node a.out.js
+
+  =================================================================
+  ==42==ERROR: LeakSanitizer: detected memory leaks
+
+  Direct leak of 40 byte(s) in 1 object(s) allocated from:
+      #0 0x5ce5 in operator new[](unsigned long)+0x5ce5 (a.out.wasm+0x5ce5)
+      #1 0xb24 in __original_main leak.cpp:2:3
+      #2 0xb3a in main+0xb3a (a.out.wasm+0xb3a)
+      #3 0x800019b8 in Object.Module._main a.out.js:6584:32
+      #4 0x80001ae1 in Object.callMain a.out.js:6881:30
+      #5 0x80001b1b in doRun a.out.js:6939:60
+      #6 0x80001b29 in run a.out.js:6953:5
+      #7 0x80001acc in runCaller a.out.js:6860:29
+
+  SUMMARY: AddressSanitizer: 40 byte(s) leaked in 1 allocation(s).
+
+Use After Return
+^^^^^^^^^^^^^^^^
+
+Consider ``use_after_return.c``:
+
+.. code-block:: c
+
+  #include <stdio.h>
+
+  const char *__asan_default_options() {
+    return "detect_stack_use_after_return=1";
+  }
+
+  int *f() {
+    int buf[10];
+    return buf;
+  }
+
+  int main() {
+    *f() = 1;
+  }
+
+Note that to do this check, you have to use the ASan option
+``detect_stack_use_after_return``. You may enable this option by declaring
+a function called ``__asan_default_options`` like the example, or you can
+define ``Module['ASAN_OPTIONS'] = 'detect_stack_use_after_return=1'`` in the
+generated JavaScript. ``--pre-js`` is helpful here.
```

Comment:
I assume this feature is off by default because it is extra expensive. It would be good to mention that if so.